sql/src/test/java/io/crate/analyze/CreateAlterTableStatementAnalyzerTest.java

```diff
@@ -186,6 +194,20 @@ public void testAlterTableWithRefreshInterval() throws Exception {
     }
 
     @Test
+    public void testAlterTableWithTotalFieldsLimit() throws Exception {
+        AlterTableAnalyzedStatement analysisSet = e.analyze(
```

Comment:
the only 3 unit tests are if the settings are set when creating the table, if the settings can be changed by alter table and if the settings can be reset. I personally think that's fine.

i'm going to rename the function and outsource the the default to its own function